src/Resources/app/administration/src/module/findologic-module/components/findologic-config/index.js

```diff
@@ -146,29 +126,14 @@ Component.register('findologic-config', {
       }
     },
 
-    /**
-     * @public
-     */
-    getCategories () {
-      this.isLoading = true;
-
-      const translatedCategories = [];
-      this.categoryRepository.search(this.categoryCriteria, Shopware.Context.api).then((items) => {
-        this.term = null;
-        this.total = items.total;
-        items.forEach((category) => {
-          translatedCategories.push({
-            value: category.id,
-            name: category.name,
-            label: category.translated.breadcrumb.join(' > ')
-          });
-        });
+    onCategoryAdd(item) {
+      this.actualConfigData['FinSearch.config.crossSellingCategories'].push(item.id)
+    },
 
-        this.categories = this.sortByProperty(translatedCategories, 'label');
-      }).finally(() => {
-        this.isLoading = false;
-      });
-    }
+    onCategoryRemove(item) {
+      this.actualConfigData['FinSearch.config.crossSellingCategories'] =
+        this.actualConfigData['FinSearch.config.crossSellingCategories'].filter(categoryId => categoryId !== item.id);
```

Comment:
`.filter()` goes though all elements. Wouldn't it be easier to just pop the element of the array?
```suggestion
        delete this.actualConfigData['FinSearch.config.crossSellingCategories'][item.id];
```

Reply:
The array is not indexed by the item id:

![image](https://user-images.githubusercontent.com/51320851/159240782-8d8cc849-b885-41dd-ab6d-2f51ffb63823.png)